src/middleware/middleware.c

```diff
@@ -51,3 +66,7 @@ void middleware_write_blocking(uint32_t address, uint8_t *data, uint16_t len) {
 uint8_t middleware_read_blocking(uint32_t address, uint8_t *data, uint16_t len) {
     qxi_read_blocking(ADDR_USER_LOGIC_OFFSET + address, data, len);
 }
+
+bool middleware_userlogic_get_busy_status(void) {
```

Comment:
What is the use case of this function?

Reply:
to check if the FPGA function is done with calculations, so we can read the result. It is used by the stub generator as per Chao's 'instructions'. This should actually check a pin connected between FPGA and MCU. The current implementation is a dummy implementation so that the stub generator produces buildable code.

Reply:
Okay. Then I'll merge it into the main branch.

Reply:
thanks. we still need an implementation of it, though. Chao send me one but I do not get it, tbh...